controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -416,7 +416,11 @@ func (r *StorageClassRequestReconciler) reconcileCephFilesystemSubVolumeGroup()
 		// This label is required to set the dataPool on the CephFS
 		// storageclass so that each PVC created from CephFS storageclass can
 		// use correct dataPool backed by deviceclass.
-		addLabel(r.cephFilesystemSubVolumeGroup, v1alpha1.CephFileSystemDataPoolLabel, fmt.Sprintf("%s-%s", cephFilesystem.Name, dataPool.Name))
+		labelValue := cephFilesystem.Name
+		if dataPool.Name != "" {
+			labelValue = fmt.Sprintf("%s-%s", labelValue, dataPool.Name)
+		}
+		addLabel(r.cephFilesystemSubVolumeGroup, v1alpha1.CephFileSystemDataPoolLabel, labelValue)
```

Comment:
we should not set the label if there is no data pool for the filesystem and we are going to use default one, and also https://github.com/red-hat-storage/ocs-operator/blob/b2c823679e4bd7c633483df193af80be11d0aced/services/provider/server/server.go#L650 this should be inside if check where set `pool` only if there is a Label set or set the default dataPool name always.

Reply:
> we should not set the label if there is no data pool for the filesystem
- ack.

> this should be inside if check where set pool only if there is a Label 
- at client side the value is just copied w/o any checks? should this be in a `if` as non-existence of label returns an empty value?

Reply:
done.